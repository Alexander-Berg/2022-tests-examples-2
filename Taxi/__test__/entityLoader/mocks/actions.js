import {actions} from './reducer';

const actionsApi = {
    testEntities: actions
};

export const pure = actionsApi;
