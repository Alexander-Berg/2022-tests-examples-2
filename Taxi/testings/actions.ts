import { createAsyncAction } from "typesafe-actions";
import { Task } from "../../types/models/Task";
import { ConfigurationTestRecord } from "../../types/models/ConfigurationTestRecord";
import { Pagination } from "../../types/models/Pagination";
import { TestingAggregation } from "../../types/models/TestingAggregation";
import { ErrorAction } from "../reducers";

export const loadTestings = createAsyncAction(
  'testings/load/requested',
  'testings/load/succeeded',
  'testings/load/failed',
)<Pagination, { tasks: Task[], pagination?: Pagination }, ErrorAction>();

export const createTesting = createAsyncAction(
  'testings/create/requested',
  'testings/create/succeeded',
  'testings/create/failed',
)<{ samplingSlug: string, useHistory?: boolean }, Task, ErrorAction>();

export const loadTestingResulsts = createAsyncAction(
  'testingResult/load/requested',
  'testingResult/load/succeeded',
  'testingResult/load/failed',
)<{ taskId: string, page: number, isEqual?: boolean }, { taskId: string, page: number, records: ConfigurationTestRecord[], total: number } , ErrorAction<{ taskId: string }>>();

export const loadTestingAggregation = createAsyncAction(
  'testings/loadAggregation/requested',
  'testings/loadAggregation/succeeded',
  'testings/loadAggregation/failed',
)<{ taskId: string }, { taskId: string, result: TestingAggregation | null }, ErrorAction<{ taskId: string }>>();
