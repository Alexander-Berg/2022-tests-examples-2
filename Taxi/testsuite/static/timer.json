{
  "data": {
    "intervals": [
      {
        "is_reverse": false,
        "mode": "descending",
        "subtitle": "test.after_eta_calculated",
        "tag": "on_route"
      },
      {
        "is_reverse": false,
        "mode": "ascending",
        "start_at": "2021-03-23T10:46:12.875501+00:00",
        "subtitle": "test.after_eta_passed",
        "tag": "on_route_late"
      }
    ],
    "subtitle": "Прибыть в ресторан"
  },
  "subtype": "timer",
  "type": "custom"
}
