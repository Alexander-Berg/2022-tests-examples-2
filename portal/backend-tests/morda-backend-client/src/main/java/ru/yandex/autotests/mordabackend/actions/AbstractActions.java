package ru.yandex.autotests.mordabackend.actions;

import ru.yandex.autotests.mordabackend.MordaClient;

import javax.ws.rs.client.Client;

/**
 * User: eoff (eoff@yandex-team.ru)
 * Date: 18.05.14
 */
public abstract class AbstractActions {
    protected MordaClient mordaClient;
    protected Client client;

    protected AbstractActions(MordaClient mordaClient, Client client) {
        this.mordaClient = mordaClient;
        this.client = client;
    }
}
