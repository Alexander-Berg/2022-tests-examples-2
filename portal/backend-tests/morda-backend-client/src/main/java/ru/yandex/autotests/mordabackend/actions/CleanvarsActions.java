package ru.yandex.autotests.mordabackend.actions;

import org.apache.commons.lang3.StringUtils;
import org.glassfish.jersey.message.internal.MessageBodyProviderNotFoundException;
import ru.yandex.autotests.mordabackend.MordaClient;
import ru.yandex.autotests.mordabackend.beans.cleanvars.Cleanvars;
import ru.yandex.autotests.mordabackend.blocks.CleanvarsBlock;
import ru.yandex.autotests.mordabackend.headers.CookieHeader;
import ru.yandex.autotests.mordabackend.http.WwwRcYandexRu;
import ru.yandex.autotests.mordabackend.useragents.UserAgent;

import javax.ws.rs.client.Client;
import java.util.List;

/**
 * User: eoff (eoff@yandex-team.ru)
 * Date: 18.05.14
 */
public class CleanvarsActions extends AbstractActions {

    public CleanvarsActions(MordaClient mordaClient, Client client) {
        super(mordaClient, client);
    }

    public String getAsString(UserAgent userAgent, List<CleanvarsBlock> cleanvarsBlocks) {
        return getAsString(null, userAgent.getValue(), null, null, cleanvarsBlocks, null);
    }

    public Cleanvars get() {
        return get(null, null, null, null, null, null);
    }

    public Cleanvars get(UserAgent userAgent) {
        return get(null, userAgent, null, null, null, null);
    }

    public Cleanvars get(UserAgent userAgent, List<CleanvarsBlock> cleanvarsBlocks) {
        return get(null, userAgent, null, null, cleanvarsBlocks, null);
    }

    public Cleanvars getWithCounters(UserAgent userAgent, List<CleanvarsBlock> cleanvarsBlocks) {
        return get(null, userAgent, null, null, cleanvarsBlocks, "1");
    }

    public Cleanvars get(UserAgent userAgent, String xForwardedFor, List<CleanvarsBlock> cleanvarsBlocks) {
        return get(null, userAgent, xForwardedFor, null, cleanvarsBlocks, null);
    }

    public Cleanvars get(CookieHeader cookieHeader) {
        return get(cookieHeader, null, null, null, null, null);
    }

    public Cleanvars get(CookieHeader cookieHeader, UserAgent userAgent) {
        return get(cookieHeader, userAgent, null, null, null, null);
    }

    public Cleanvars get(CookieHeader cookieHeader, UserAgent userAgent, List<CleanvarsBlock> cleanvarsBlocks) {
        return get(cookieHeader, userAgent, null, null, cleanvarsBlocks, null);
    }

    public Cleanvars get(CookieHeader cookieHeader, UserAgent userAgent, String xForwardedFor) {
        return get(cookieHeader, userAgent, xForwardedFor, null, null, null);
    }

    public Cleanvars get(CookieHeader cookieHeader, UserAgent userAgent, String xForwardedFor,
                         String referer, List<CleanvarsBlock> blocks, String forcecounters)
    {
        return get(cookieHeader, userAgent == null ? null : userAgent.getValue(),
                xForwardedFor, referer, blocks, forcecounters, true);
    }

    public Cleanvars get(CookieHeader cookieHeader, String userAgent, String xForwardedFor,
                         String referer, List<CleanvarsBlock> blocks, String forcecounters, boolean notUsed)
    {
        try {
            return getImpl(cookieHeader, userAgent, xForwardedFor, referer, blocks, forcecounters);
        } catch (MessageBodyProviderNotFoundException e) {
            try {
                return getImpl(cookieHeader, userAgent, xForwardedFor, referer, blocks, forcecounters);
            } catch (MessageBodyProviderNotFoundException e2) {
                return getImpl(cookieHeader, userAgent, xForwardedFor, referer, blocks, forcecounters);
            }
        }
    }

    private Cleanvars getImpl(CookieHeader cookieHeader, String userAgent, String xForwardedFor,
                              String referer, List<CleanvarsBlock> blocks, String forcecounters) {
        return WwwRcYandexRu.root(client, mordaClient.getMordaHost())
                .getAsJson(
                        (blocks == null || blocks.size() == 0) ? "1" : "^" + StringUtils.join(blocks, "$|^") + "$",
                        forcecounters,
                        userAgent,
                        cookieHeader == null ? null : cookieHeader.toString(),
                        xForwardedFor,
                        referer,
                        Cleanvars.class
                );
    }

    private String getAsString(CookieHeader cookieHeader, String userAgent, String xForwardedFor,
                              String referer, List<CleanvarsBlock> blocks, String forcecounters) {
        return WwwRcYandexRu.root(client, mordaClient.getMordaHost())
                .getAsJson(
                        (blocks == null || blocks.size() == 0) ? "1" : "^" + StringUtils.join(blocks, "$|^") + "$",
                        forcecounters,
                        userAgent,
                        cookieHeader == null ? null : cookieHeader.toString(),
                        xForwardedFor,
                        referer,
                        String.class
                );
    }

    public void setLanguage(String lang, String sk) {
        WwwRcYandexRu.root(client, mordaClient.getMordaHost())
                .get(lang, sk).close();
    }
}
