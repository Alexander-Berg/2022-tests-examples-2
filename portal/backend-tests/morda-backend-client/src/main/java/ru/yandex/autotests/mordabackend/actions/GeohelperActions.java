package ru.yandex.autotests.mordabackend.actions;

import ru.yandex.autotests.mordabackend.MordaClient;
import ru.yandex.autotests.mordabackend.beans.geohelper.GeohelperResponse;
import ru.yandex.autotests.mordabackend.beans.poi_groups.PoiGroupsItem;

import javax.ws.rs.client.Client;
import javax.ws.rs.client.Entity;
import javax.ws.rs.core.MediaType;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * @author Ivan Nikolaev <ivannik@yandex-team.ru>
 */
public class GeohelperActions extends AbstractActions {

    public GeohelperActions(MordaClient mordaClient, Client client) {
        super(mordaClient, client);
    }

    public GeohelperResponse getGeohelperResponse(String lat, String lon, String geoid, List<PoiGroupsItem> poiGroups) {
        Map<String, List<PoiGroupsItem>> postPoiData = new HashMap<>();
        postPoiData.put("poi", poiGroups);

        String mordaHost = mordaClient.getMordaHost().toString().replaceAll("(?<=//).+(?=\\.yandex)","l7test");

        return client
                .target(mordaHost)
                .path("/geohelper/get")
                .queryParam("lat", lat)
                .queryParam("lon", lon)
                .queryParam("geoid", geoid)
                .request()
                .post(Entity.entity(postPoiData, MediaType.APPLICATION_JSON_TYPE), GeohelperResponse.class);
    }
}
