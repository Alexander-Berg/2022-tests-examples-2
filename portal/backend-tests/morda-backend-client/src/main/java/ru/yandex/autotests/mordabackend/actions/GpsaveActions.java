package ru.yandex.autotests.mordabackend.actions;

import com.fasterxml.jackson.databind.JsonNode;
import ru.yandex.autotests.mordabackend.MordaClient;

import javax.ws.rs.client.Client;

/**
 * @author Ivan Nikolaev <ivannik@yandex-team.ru>
 */
public class GpsaveActions extends AbstractActions {

    public GpsaveActions(MordaClient mordaClient, Client client) {
        super(mordaClient, client);
    }

    public JsonNode gpsave(String lat, String lon, String precision, String sk) {
        return action("gpsave", lat, lon, precision, sk);
    }

    public JsonNode mgpsave(String lat, String lon, String precision, String sk) {
        return action("mgpsave", lat, lon, precision, sk);
    }

    private JsonNode action(String path, String lat, String lon, String precision, String sk) {
        return client
                .target(mordaClient.getMordaHost())
                .path(path)
                .queryParam("lat", lat)
                .queryParam("lon", lon)
                .queryParam("precision", precision)
                .queryParam("sk", sk)
                .request()
                .get(JsonNode.class);
    }
}
