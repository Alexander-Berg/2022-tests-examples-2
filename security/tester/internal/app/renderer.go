package app

import (
	"bytes"
	_ "embed"
	"fmt"
	"io"
	"strings"
	"time"

	"golang.org/x/crypto/ssh"
	"golang.org/x/crypto/ssh/agent"

	"a.yandex-team.ru/security/skotty/skotty/pkg/sshutil"
)

func (a *App) renderClientInfo(cl *ClientInfo) ([]byte, error) {
	formatBool := func(b bool) string {
		if b {
			return "yes"
		}
		return "no"
	}

	staffedKey := func(key ssh.PublicKey) string {
		if cl.StaffkeysErr != nil {
			return fmt.Sprintf("unble to check: %s", cl.StaffkeysErr)
		}

		var target string
		if cert, ok := key.(*ssh.Certificate); ok {
			target = ssh.FingerprintSHA256(cert.SignatureKey)
		} else {
			target = ssh.FingerprintSHA256(key)
		}

		for _, sk := range cl.StaffKeys {
			if sk == target {
				return formatBool(true)
			}
		}

		return formatBool(false)
	}

	formatKey := func(out io.Writer, in ssh.PublicKey) {
		writeCertInfo := func(key ssh.PublicKey) {
			crt, ok := key.(*ssh.Certificate)
			if !ok {
				return
			}

			_, _ = fmt.Fprintf(out, "      - revoked: %s\n", formatBool(a.krl.IsRevoked(crt)))
			_, _ = fmt.Fprintf(out, "      - principals: %s\n", strings.Join(crt.ValidPrincipals, ","))
			_, _ = fmt.Fprintf(out, "      - valid before: %s\n", time.Unix(int64(crt.ValidBefore), 0).Format("2006-01-02T15:04"))
			_, _ = fmt.Fprintf(out, "      - ca fingerprint: %s\n", sshutil.Fingerprint(crt.SignatureKey))
			_, _ = fmt.Fprintf(out, "      - serial: %d\n", crt.Serial)
			_, _ = fmt.Fprintf(out, "      - key id: %s\n", crt.KeyId)
		}

		writeKeyInfo := func(key ssh.PublicKey) {
			_, _ = fmt.Fprintf(out, "    * %s\n", sshutil.Fingerprint(key))
			_, _ = fmt.Fprintf(out, "      - type: %s\n", key.Type())
			_, _ = fmt.Fprintf(out, "      - present on staff: %s\n", staffedKey(key))
			writeCertInfo(key)
		}

		switch v := in.(type) {
		case *agent.Key:
			sshKey, err := ssh.ParsePublicKey(v.Marshal())
			if err != nil {
				break
			}

			writeKeyInfo(sshKey)
			_, _ = fmt.Fprintf(out, "      - comment: %s\n", v.Comment)
		default:
			writeKeyInfo(v)
		}
	}

	var out bytes.Buffer
	_, _ = fmt.Fprint(&out, "Welcome to the Skotty Validation server!\n\n")

	if cl.ClientCompatNotice != "" {
		_, _ = fmt.Fprintln(&out, cl.ClientCompatNotice)
	}
	_, _ = fmt.Fprintln(&out, "Info about your connection:")
	_, _ = fmt.Fprintf(&out, "  - Client version: %s\n", cl.ClientVersion)
	_, _ = fmt.Fprintf(&out, "  - Username: %s\n", cl.Username)
	_, _ = fmt.Fprintf(&out, "  - Auth keys: %d\n", len(cl.AuthKeys))
	for _, key := range cl.AuthKeys {
		formatKey(&out, key)
	}

	if cl.AgentForwarded {
		_, _ = fmt.Fprintln(&out, "  - Agent forwarded: yes")
		_, _ = fmt.Fprintf(&out, "  - Agent support sudo: %s\n", formatBool(cl.SudoAvailable))
		_, _ = fmt.Fprintf(&out, "  - Agent keys: %d\n", len(cl.AgentKeys))
		for _, key := range cl.AgentKeys {
			formatKey(&out, key)
		}
	} else {
		_, _ = fmt.Fprintln(&out, "  - Agent forwarded: no")
	}

	_ = out.WriteByte('\n')
	_, _ = out.WriteString("Have a great day! ;)\n")
	return out.Bytes(), nil
}
