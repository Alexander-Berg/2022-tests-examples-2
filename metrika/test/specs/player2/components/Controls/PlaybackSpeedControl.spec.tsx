import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import ActionsWrapper from '@test/utils/ActionsWrapper'
import { PlaybackSpeedControl } from '@player2/components/Controls/PlaybackSpeedControl'
import MainThemeProvider from '@player2/components/Themes/MainThemeProvider'
import { setSpeed } from '@player2/store/actions/timer'
import { DEFAULT_PLAY_SPEED } from '@player2/store/reducers/timer'
import { mount } from 'enzyme'

describe('<PlaybackSpeedControl>', () => {
  const actionsWrapper = new ActionsWrapper({
    setSpeed,
  })
  const actionsProps = actionsWrapper.getProxiedActions()
  const props = {
    x1: {
      speed: DEFAULT_PLAY_SPEED,
      ...actionsProps,
    },
    x2: {
      speed: DEFAULT_PLAY_SPEED * 2,
      ...actionsProps,
    },
    x3: {
      speed: DEFAULT_PLAY_SPEED * 3,
      ...actionsProps,
    },
    x4: {
      speed: DEFAULT_PLAY_SPEED * 4,
      ...actionsProps,
    },
  } as any
  it ('renders correctly', () => {
    let tree = renderer.create(<MainThemeProvider><PlaybackSpeedControl {...props.x1}/></MainThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
    tree = renderer.create(<MainThemeProvider><PlaybackSpeedControl {...props.x2}/></MainThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
    tree = renderer.create(<MainThemeProvider><PlaybackSpeedControl {...props.x3}/></MainThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
    tree = renderer.create(<MainThemeProvider><PlaybackSpeedControl {...props.x4}/></MainThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it  ('calls actions correctly', () => {
    actionsWrapper.clear()
    const element = mount(<MainThemeProvider><PlaybackSpeedControl {...props.x1}/></MainThemeProvider>)
    element.find('button').simulate('click')
    element.setProps({children: <PlaybackSpeedControl {...props.x2}/>})
    element.find('button').simulate('click')
    element.setProps({children: <PlaybackSpeedControl {...props.x3}/>})
    element.find('button').simulate('click')
    element.setProps({children: <PlaybackSpeedControl {...props.x4}/>})
    element.find('button').simulate('click')

    expect(actionsWrapper.getCalledActions()).toEqual([
      setSpeed(DEFAULT_PLAY_SPEED * 2),
      setSpeed(DEFAULT_PLAY_SPEED * 3),
      setSpeed(DEFAULT_PLAY_SPEED * 4),
      setSpeed(DEFAULT_PLAY_SPEED),
    ])
  })
})
