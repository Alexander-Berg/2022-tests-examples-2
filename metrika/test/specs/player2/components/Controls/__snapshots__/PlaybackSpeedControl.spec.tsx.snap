// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaybackSpeedControl> renders correctly 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  border: none;
  background: none;
  outline: none;
  cursor: pointer;
}

.c1 svg {
  display: block;
}

.c0 {
  width: 55px;
  height: 55px;
  opacity: 0.6;
}

.c0 svg {
  width: 100%;
  height: 100%;
  fill: rgba(255,255,255,1);
}

.c0:hover {
  opacity: 0.8;
}

.c0:active {
  opacity: 0.75;
}

<button
  className="c0 c1"
  onClick={[Function]}
/>
`;

exports[`<PlaybackSpeedControl> renders correctly 2`] = `
.c1 {
  margin: 0;
  padding: 0;
  border: none;
  background: none;
  outline: none;
  cursor: pointer;
}

.c1 svg {
  display: block;
}

.c0 {
  width: 55px;
  height: 55px;
  opacity: 0.6;
}

.c0 svg {
  width: 100%;
  height: 100%;
  fill: rgba(255,255,255,1);
}

.c0:hover {
  opacity: 0.8;
}

.c0:active {
  opacity: 0.75;
}

<button
  className="c0 c1"
  onClick={[Function]}
/>
`;

exports[`<PlaybackSpeedControl> renders correctly 3`] = `
.c1 {
  margin: 0;
  padding: 0;
  border: none;
  background: none;
  outline: none;
  cursor: pointer;
}

.c1 svg {
  display: block;
}

.c0 {
  width: 55px;
  height: 55px;
  opacity: 0.6;
}

.c0 svg {
  width: 100%;
  height: 100%;
  fill: rgba(255,255,255,1);
}

.c0:hover {
  opacity: 0.8;
}

.c0:active {
  opacity: 0.75;
}

<button
  className="c0 c1"
  onClick={[Function]}
/>
`;

exports[`<PlaybackSpeedControl> renders correctly 4`] = `
.c1 {
  margin: 0;
  padding: 0;
  border: none;
  background: none;
  outline: none;
  cursor: pointer;
}

.c1 svg {
  display: block;
}

.c0 {
  width: 55px;
  height: 55px;
  opacity: 0.6;
}

.c0 svg {
  width: 100%;
  height: 100%;
  fill: rgba(255,255,255,1);
}

.c0:hover {
  opacity: 0.8;
}

.c0:active {
  opacity: 0.75;
}

<button
  className="c0 c1"
  onClick={[Function]}
/>
`;
